uhal/tools/scripts/gen_ipbus_addr_decode

```diff
@@ -82,178 +70,216 @@ class BitArray:
+        l = list()
+        for i in n.desc:
+            if i.flag:
+                l.append(i)
+            else:
+                l.extend(self.get_flagged_nodes(i))
         return l
 
 #===========================================================================================
 
-EXIT_CODE_INCORRECT_ARGUMENTS = 1
-EXIT_CODE_ARG_PARSING_ERROR   = 2
-EXIT_CODE_NODE_ADDRESS_ERRORS = 3
+EXIT_CODE_ARG_PARSING_ERROR   = 1
+EXIT_CODE_NODE_ADDRESS_ERRORS = 2
 
 def main():    
+    logging.basicConfig(level=logging.WARNING, format='%(levelname)s\t: %(message)s')
 
     # configure logger
     global log
     log = logging.getLogger("main")
-    formatter = logging.Formatter('%(name)s %(levelname)s: %(message)s')
-    handler = logging.StreamHandler(sys.stdout)
-    handler.setFormatter(formatter)
-    log.addHandler(handler)
-    log.setLevel(logging.WARNING)
+
     uhal.setLogLevelTo(uhal.LogLevel.WARNING)
 
     # options for verbosity level and VHDL template
     nflag = False
     template_fn = "/opt/cactus/etc/uhal/tools/ipbus_addr_decode.vhd"
-    try:
-        opts, args = getopt.getopt(sys.argv[1:], "vdnht:", ["verbose","debug","dry-run","help","template="])
-    except getopt.GetoptError as err:
-        log.critical(__doc__)
-        sys.exit(EXIT_CODE_ARG_PARSING_ERROR)
-    for o, a in opts:
-        if o in ("-v", "--verbose"):
-            log.setLevel(logging.INFO)
-            uhal.setLogLevelTo(uhal.LogLevel.INFO)
-        elif o in ("-d", "--debug"):
-            log.setLevel(logging.DEBUG)
-            uhal.setLogLevelTo(uhal.LogLevel.DEBUG)
-        elif o in ("-h", "--help"):
-            print(__doc__)
-            sys.exit(0)
-        elif o in ("-t", "--template"):
-            template_fn = a
-        elif o in ("-n", "--dry-run"):
-            nflag = True
-
-    # make sure that exactly one argument was given, later assumed to be the xml file name
-    if len(args) != 1:
-        log.critical("Incorrect usage - invalid number of arguments! Make sure that options come before argument.\n" + __doc__)
-        sys.exit(EXIT_CODE_INCORRECT_ARGUMENTS)
 
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter
+        )
+    parser.add_argument('-v', '--verbose', help="increase output verbosity (default: %(default)s)", action="store_true")
+    parser.add_argument('-d', '--debug', help="enable debug messages (default: %(default)s)", action="store_true")
+    parser.add_argument('-n', '--dry-run', help="Dry run (default: %(default)s)", action="store_true", default=False)
+    parser.add_argument('-t', '--template', help="Decoder template path (default: %(default)s)", default='/opt/cactus/etc/uhal/tools/ipbus_addr_decode.vhd')
+    parser.add_argument('addrtab', help="Address table file")
+    args = parser.parse_args()
+
+    if args.verbose:
+        log.setLevel(logging.INFO)
+        uhal.setLogLevelTo(uhal.LogLevel.INFO)
+    if args.verbose:
```

Comment:
TODO: change `args.verbose` to `args.debug` here

Reply:
Done.